src/bio/bio_xstream.c

```diff
@@ -116,6 +117,46 @@ opts_add_pci_addr(struct spdk_env_opts *opts, struct spdk_pci_addr **list,
 	return 0;
 }
 
+/*
+ * Convert a transport id in the BDF form of "5d0505:01:00.0" or something
+ * similar to the VMD address in the form of "0000:5d:05.5" that can be parsed
+ * by DPDK.
+ */
```

Comment:
What if we just trim the VMD address from whitelist? I'm still don't quite understand the purpose of this whitelist.

Reply:
We have to pass a whitelist with NVMe devices for DPDK (correct me if I am wrong about this @tanabarr). So if there are regular (non-VMD) NVMe SSDs they will have to be passed in the whitelist, and then the VMD address would have to be passed as well or it will not be used. If it is JUST VMD devices, I assume we can not pass a whitelist at all, however seems the safer bet to just pass the VMD address (which will include all of the NVMe SSDs behind the VMD). I don't know a way to specify only specific drives behind the VMD for the whitelist, expect manually in the BIOS.